### PR TITLE
Add more datapack API

### LINKED ISSUE #4654: Load new data pack from file (directory or zip)
**Is your feature request related to a problem? Please describe.**
This feature is great for custom loot table and provide it by plugin and not by default world.

**Describe the solution you'd like**
An method like `Bukkit.loadDataPack(NamespacedKey, File)` (first is key of data pack and second if file or data pack).

**Describe alternatives you've considered**
I have try to solve the loot table solution via NMS but this is really hard to make it.
Also i think just an method like `Bukkit.loadLootTable(NamespacedKey, File)` (with file is the json of loot table), could be great.

### PR DESCRIPTION
**Pull request by NoahvdAa**
Adds a new `DatapackManager#loadDatapack` method, allowing users to load datapacks from the API.

Fixes #4654.

### CODE REVIEW

**Comment by NoahvdAa** [comment]
Rebased for 1.18!

**Comment by NoahvdAa** [comment]
Rebased for 1.18.2, label can be removed.

File: patches/api/0296-Datapack-API.patch [hunk 1]
```diff
@@ -68,6 +94,31 @@ index 0000000000000000000000000000000000000000..58f78d5e91beacaf710f62461cf869f7
...
++     * @throws IOException if there was an error while reading the pack file
++     * @return the newly loaded pack
++     */
++    @Nullable
++    default Datapack loadDatapack(@NotNull String name, @NotNull File file) throws DatapackFileNotFoundException, IOException {
++        return loadDatapack(name, file, true);
++    }
++
++    /**
++     * Loads a new datapack and enables it.
```
Comment (Machine-Maker) [inline patches/api/0296-Datapack-API.patch]:
```suggestion
+     * Loads a new datapack and optionally enables it.
```

File: patches/api/0296-Datapack-API.patch [hunk 2]
```diff
@@ -42,16 +43,41 @@ index 0000000000000000000000000000000000000000..7b2ab0be10a21e0496ad1d485ff8cb2c
...
++import java.io.FileNotFoundException;
++
++/**
++ * Thrown when the server can't load a datapack because a file is missing.
++ */
++public class DatapackFileNotFoundException extends FileNotFoundException {
++    public DatapackFileNotFoundException(@NotNull File packSource, @NotNull String resource) {
++        super(String.format("'%s' in DataPack '%s'", resource, packSource));
++    }
++}
```
Comment (Machine-Maker) [inline patches/api/0296-Datapack-API.patch]: Do we need a custom exception? All its doing is a custom string.format, can't that just be done in the FileNotFoundException ctor?

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by NoahvdAa** [comment]
Thanks, fixed

**Pull request closed without merging**